src/tests/componenet.test.js

```diff
@@ -0,0 +1,22 @@
+import { render, cleanup } from '@testing-library/react';
+import { Provider } from 'react-redux';
+import { store } from '../Redux/store';
+import Filter from '../components/input/filter/Filter';
+import Search from '../components/input/search/Search';
+import Country from '../components/country/Country';
+
+afterEach(cleanup);
+describe('tests for components', () => {
+  test('Snapshot test for filter', () => {
+    const countryfilter = render(<Provider store={store}><Filter /></Provider>);
+    expect(countryfilter).toMatchSnapshot();
+  });
+  test('Snapshot test for Country', () => {
+    const countryDipslayed = render(<Provider store={store}><Country /></Provider>);
+    expect(countryDipslayed).toMatchSnapshot();
+  });
+  test('Snapshot test for Search', () => {
+    const countrySearched = render(<Provider store={store}><Search /></Provider>);
+    expect(countrySearched).toMatchSnapshot();
+  });
+});
```

Comment:
- [ ] Again, everything is working well, however I want to mention that testing is also a crucial part in development, currently you have written test cases for components only, as mentioned in the project requirements :point_down: 
![image](https://user-images.githubusercontent.com/97521557/233274592-65c95f90-e61a-42dd-bad5-a2af60972f28.png)

Please try to add test cases for reducers and actions, I will try to provide some steps you can take below

- Import the necessary dependencies for testing: redux-mock-store, thunk, and jest.
- Create a mock store using the redux-mock-store library.
- Import the reducer and the reducer action creators to be tested.

example snippet : 
import dependencies needed

```
const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('countriesSlice reducers', () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      loading: false,
      countriesData: [],
      countrySearched: [],
      error: false,
      searchTerm: '',
      success: false,
      message: '',
    });
  });

  it('should set the search term', () => {
    const searchTerm = 'test';
    const expectedAction = {
      type: 'countries/setSearchTerm',
      payload: searchTerm,
    };

    store.dispatch(setSearchTerm(searchTerm));
    expect(store.getActions()).toEqual([expectedAction]);
  });
});

```